src/AltinnCore/Common/Services/Implementation/DataAppSI.cs

```diff
@@ -119,21 +121,112 @@ public object GetFormData(Guid instanceId, Type type, string applicationOwner, s
         }
 
         /// <inheritdoc />
-        public List<AttachmentList> GetFormAttachments(string applicationOwnerId, string applicationId, int instanceOwnerId, Guid instanceId)
+        public async Task<List<AttachmentList>> GetFormAttachments(string applicationOwnerId, string applicationId, int instanceOwnerId, Guid instanceId)
         {
-            throw new NotImplementedException();
+            List<Data> dataList = null;
+            List<AttachmentList> attachmentList = new List<AttachmentList>();
+            List<Attachment> attachments = null;
+            string apiUrl = $"{_platformStorageSettings.ApiUrl}/instances/{instanceId}/data?instanceOwnerId={instanceOwnerId}";
+            using (HttpClient client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(apiUrl);
+
+                HttpResponseMessage response = await client.GetAsync(apiUrl);
+                if (response.StatusCode == System.Net.HttpStatusCode.OK)
+                {
+                    string instanceData = await response.Content.ReadAsStringAsync();
+                    dataList = JsonConvert.DeserializeObject<List<Data>>(instanceData);
+
+                    IEnumerable<Data> attachmentTypes = dataList.GroupBy(m => m.FormId).Select(m => m.FirstOrDefault());
+
+                    foreach (Data attachmentType in attachmentTypes)
+                    {
+                        attachments = new List<Attachment>();
+                        foreach (Data data in dataList)
+                        {
+                            if (data.FormId != "default" && data.FormId == attachmentType.FormId)
+                            {
+                                attachments.Add(new Attachment
+                                {
+                                    Id = data.Id,
+                                    Name = data.FileName,
+                                    Size = data.FileSize
+                                });
+                            }
+                        }
+
+                        if (attachments.Count > 0)
+                        {
+                            attachmentList.Add(new AttachmentList { Type = attachmentType.FormId, Attachments = attachments });
+                        }
+                    }
+
+                    if (attachments.Count > 0)
+                    {
+                        attachmentList.Add(new AttachmentList { Type = "attachments", Attachments = attachments });
+                    }
+
+                }
+                else
+                {
+                    throw new Exception("Unable to fetch attachment list");
```

Comment:
Isnt it better to just log that one is unable to fetch attachment list and return an empty list? Throwing errors in code is maybe not the best solution

Reply:
Fixed